web/src/emoji_picker.ts

```diff
@@ -287,44 +334,48 @@ function toggle_reaction(emoji_name, event) {
     }
 
     const message_id = current_message_id;
-    const message = message_store.get(message_id);
-    if (!message) {
-        blueslip.error("reactions: Bad message id", {message_id});
-        return;
-    }
+    if (message_id) {
+        const message = message_store.get(message_id);
+        if (!message) {
+            blueslip.error("reactions: Bad message id", {message_id});
+            return;
+        }
 
-    reactions.toggle_emoji_reaction(message, emoji_name, event);
+        reactions.toggle_emoji_reaction(message, emoji_name);
 
-    if (!event.shiftKey) {
-        hide_emoji_popover();
-    }
+        if (!event.shiftKey) {
+            hide_emoji_popover();
+        }
 
-    $(event.target).closest(".reaction").toggleClass("reacted");
+        $(event.target).closest(".reaction").toggleClass("reacted");
+    }
 }
 
-function process_enter_while_filtering(e) {
+function process_enter_while_filtering(e: JQuery.KeyboardEventBase): void {
     if (keydown_util.is_enter_event(e)) {
         e.preventDefault();
         e.stopPropagation();
         const $first_emoji = get_rendered_emoji(0, 0);
         if ($first_emoji) {
-            handle_emoji_clicked($first_emoji, e);
+            // eslint-disable-next-line new-cap
+            const clickEvent: JQuery.ClickEvent = $.Event("click");
+            handle_emoji_clicked($first_emoji, clickEvent);
```

Comment:
This seems like a weird hack; why not just make `handle_emoji_clicked` except a `JQuery.KeyboardEventBase | JQuery.ClickEvent`?

Reply:
The only thing the event parameter looks to be used for is checking `event.shiftKey` in `toggle_reaction` -- so definitely wrong to replace it with a fake click event.